Assets/BossRoom/Scripts/Client/UI/UIQuitPanel.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Unity.Multiplayer.Samples.BossRoom;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Client
+{
+    public class UIQuitPanel : MonoBehaviour
+    {
+        [SerializeField]
+        Text m_QuitButtonText;
+
+        void OnEnable()
+        {
+            m_QuitButtonText.text = NetworkManager.Singleton != null && NetworkManager.Singleton.IsListening ?
+                "Leave session?" :
+                "Exit Game?";
+        }
+
+        public void Quit()
+        {
+            if (NetworkManager.Singleton.IsListening)
+            {
+                // first disconnect then return to menu
+                var gameNetPortal = GameObject.FindGameObjectWithTag("GameNetPortal").GetComponent<GameNetPortal>();
```

Comment:
GameNetPortal has a static "Instance".
Finding objects with tags is a maintenance nightmare (can't "find all references" on those)

Reply:
Didn't see it had a static instance, I'll switch to that. I didn't like using tags for that, but from what I understood it was better than FindObjectOfType() in terms of performance.